src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -137,67 +186,148 @@ Json::Value CascadiaSettings::ToJson() const
-            }
+            _LayerOrCreateProfile(profileJson);
         }
     }
+}
 
-    return resultPtr;
+// Method Description:
+// - Given a partial json serialization of a Profile object, either layers that
+//   json on a matching Profile we already have, or creates a new Profile
+//   object from those settings.
+// Arguments:
+// - json: an object which may be a partial serialization of a Profile object.
+// Return Value:
+// - <none>
+void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
+{
+    // Layer the json on top of an existing profile, if we have one:
+    auto pProfile = _FindMatchingProfile(profileJson);
+    if (pProfile)
+    {
+        pProfile->LayerJson(profileJson);
+    }
+    else
+    {
+        auto profile = Profile::FromJson(profileJson);
+        _profiles.emplace_back(profile);
+    }
+}
+
+// Method Description:
+// - Finds a profile from our list of profiles that matches the given json
+//   object. Uses Profile::ShouldBeLayered to determine if the Json::Value is a
+//   match or not. This method should be used to find a profile to layer the
+//   given settings upon.
+// - Returns nullptr if no such match exists.
+// Arguments:
+// - <none>
+// Return Value:
+// - a Profile that can be layered with the given json object, iff such a
+//   profile exists.
+Profile* CascadiaSettings::_FindMatchingProfile(const Json::Value& profileJson)
+{
+    for (auto& profile : _profiles)
+    {
+        if (profile.ShouldBeLayered(profileJson))
+        {
+            return &profile;
+        }
+    }
+    return nullptr;
+}
+
+// Method Description:
+// - Given a partial json serialization of a ColorScheme object, either layers that
+//   json on a matching ColorScheme we already have, or creates a new ColorScheme
+//   object from those settings.
+// Arguments:
+// - json: an object which should be a partial serialization of a ColorScheme object.
+// Return Value:
+// - <none>
+void CascadiaSettings::_LayerOrCreateColorScheme(const Json::Value& schemeJson)
+{
+    // Layer the json on top of an existing profile, if we have one:
+    auto pScheme = _FindMatchingColorScheme(schemeJson);
+    if (pScheme)
+    {
+        pScheme->LayerJson(schemeJson);
+    }
+    else
+    {
+        auto scheme = ColorScheme::FromJson(schemeJson);
+        _globals.GetColorSchemes().emplace_back(scheme);
+    }
+}
+
+// Method Description:
+// - Finds a color scheme from our list of color schemes that matches the given
+//   json object. Uses ColorScheme::ShouldBeLayered to determine if the
+//   Json::Value is a match or not. This method should be used to find a color
+//   scheme to layer the given settings upon.
+// - Returns nullptr if no such match exists.
+// Arguments:
+// - <none>
+// Return Value:
+// - a ColorScheme that can be layered with the given json object, iff such a
+//   color scheme exists.
+ColorScheme* CascadiaSettings::_FindMatchingColorScheme(const Json::Value& schemeJson)

```

Comment:
This probably should have the same concern/resolution as Dustin's comment above.